pyi.py

```diff
@@ -187,6 +239,76 @@ def in_class(self) -> bool:
         """Determine whether we are inside a `class` statement"""
         return bool(self._class_nesting)
 
+    def _check_typing_object(
+        self, node: ast.Attribute | ast.ImportFrom, module_name: str, object_name: str
+    ) -> None:
+        if object_name in _BAD_COLLECTIONS_ALIASES:
+            error_code, blacklist = Y022, _BAD_COLLECTIONS_ALIASES
+        elif object_name == "AsyncContextManager":
+            error_code = Y022
+            blacklist = {
+                "AsyncContextManager": '"contextlib.AbstractAsyncContextManager"'
+            }
+        elif module_name == "typing":
+            if object_name not in _BAD_BUILTINS_ALIASES:
+                return
+            error_code, blacklist = Y022, _BAD_BUILTINS_ALIASES
+        elif object_name in _BAD_COLLECTIONS_ABC_ALIASES:
+            error_code, blacklist = Y023, _BAD_COLLECTIONS_ABC_ALIASES
+        elif object_name in _TYPING_NOT_TYPING_EXTENSIONS:
+            error_code, blacklist = Y023, _TYPING_NOT_TYPING_EXTENSIONS
+        elif object_name == "ContextManager":
+            error_code = Y023
+            blacklist = {
+                "ContextManager": '"contextlib.AbstractContextManager" or "typing.ContextManager"'
+            }
+        else:
+            return
+
+        error_message = error_code.format(
+            good_cls_name=blacklist[object_name],
+            bad_cls_alias=f"{module_name}.{object_name}",
+        )
+        self.error(node, error_message)
+
+    def visit_Attribute(self, node: ast.Attribute) -> None:
+        self.generic_visit(node)
+        thing = node.value
+        if not isinstance(thing, ast.Name):
+            return
+        thingname, attribute = thing.id, node.attr
```

Comment:
```suggestion
        module_name, attribute = thing.id, node.attr
```
More precise variable name

Reply:
I went for `thingname` because we *don't* know that it's a module here -- this method visits all `ast.Attribute` nodes. It's only later down the method that we narrow it down as to whether it's an `ast.Attribute` we're interested in or not.